src/github.com/matrix-org/dendrite/clientapi/routing/joinroom.go

```diff
@@ -207,13 +207,17 @@ func (r joinRoomReq) writeToBuilder(eb *gomatrixserverlib.EventBuilder, roomID s
 func (r joinRoomReq) joinRoomUsingServers(
 	roomID string, servers []gomatrixserverlib.ServerName,
 ) util.JSONResponse {
+	evTime, resErr := httputil.ParseTSParam(r.req)
+	if resErr != nil {
+		return *resErr
```

Comment:
Re https://github.com/matrix-org/dendrite/pull/576/files#discussion_r208269878

Here you have an error, yet you return it as the response.
This is okay, since the error has been handled and the caller doesn't need to.

With similar reasoning you could argue the you can just return `resErr, nil`

Reply:
the ts-param parsing is now done before this method gets called. This is more clear and avoids having this confusion with MatrixError (from the linked discussion)